src/HttpCache/EventListener/AddHeadersListener.php

```diff
@@ -30,16 +31,21 @@
     private $sharedMaxAge;
     private $vary;
     private $public;
+    private $resourceMetadataFactory;
 
-    public function __construct(bool $etag = false, int $maxAge = null, int $sharedMaxAge = null, array $vary = null, bool $public = null)
+    public function __construct(ResourceMetadataFactoryInterface $resourceMetadataFactory, bool $etag = false, int $maxAge = null, int $sharedMaxAge = null, array $vary = null, bool $public = null)
     {
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
         $this->etag = $etag;
         $this->maxAge = $maxAge;
         $this->sharedMaxAge = $sharedMaxAge;
         $this->vary = $vary;
         $this->public = $public;
     }
 
+    /**
+     * @throws \ApiPlatform\Core\Exception\ResourceClassNotFoundException
```

Comment:
I think that `@throws` tags are not propagated to parent function in the core usually